packages/core/src/external/carequality/ihe-gateway-v2/outbound/xca/mtom/parser.ts

```diff
@@ -113,22 +119,27 @@ export function parseMTOMResponse(mtomMessage: string, contentType: string): Doc
         removeNSPrefix: true,
       });
       const jsonObj = parser.parse(content);
+      console.log("jsonObj", JSON.stringify(jsonObj, null, 2));
 
       const docResponses = Array.isArray(
         jsonObj?.Envelope?.Body?.RetrieveDocumentSetResponse?.DocumentResponse
       )
         ? jsonObj.Envelope.Body.RetrieveDocumentSetResponse.DocumentResponse
         : [jsonObj.Envelope.Body.RetrieveDocumentSetResponse.DocumentResponse];
       for (const docResponse of docResponses) {
-        // temporarily skip non-xml documents
-        if (
-          docResponse.mimeType === XML_APP_MIME_TYPE ||
-          docResponse.mimeType === XML_TXT_MIME_TYPE
-        ) {
-          documentResponses.push({
-            ...docResponse,
-            Document: stripCidPrefix(docResponse.Document.Include._href),
-          });
+        if (containsMultipartCidReference(docResponse)) {
```

Comment:
adding handling for the case where response is MTOM but document is still in SOAP. This is how athena sends